habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -821,7 +822,14 @@ def step(self, action: Union[str, int]) -> Observations:
 
         self.maybe_update_articulated_agent()
 
-        if self._concur_render:
+        if self._batch_render:
+            for _ in range(self.ac_freq_ratio):
+                self.internal_step(-1, update_articulated_agent=False)
+
+            self._prev_sim_obs = self.get_sensor_observations()
+            self.add_keyframe_to_observations(self._prev_sim_obs)
+            obs = self._prev_sim_obs
+        elif self._concur_render:
             self._prev_sim_obs = self.start_async_render()
```

Comment:
I don't understand this line (and I know this is just existing code). As far as I can see, `start_async_render` doesn't return anything. So is this code setting `_prev_sim_obs` to None? And the third case (not batch rendering and not concurrent rendering) doesn't update `_prev_sim_obs` at all? And so it's not clear to me what your code should do. Can you ask @ASzot what we should be doing with `_prev_sim_obs` in here? He recently made [this change](https://github.com/facebookresearch/habitat-lab/pull/1331/files#diff-2e30591431a2b62e1c995e1d0af87b17231e46f20c4fb892ad4ee2a0bc8e4aebL761).

Reply:
Right, this is odd. The only place where `_prev_sim_obs` is used is to return `previous_step_collided()` (within the parent object `HabitatSim`).

Let me see what's the status on this.

Reply:
@ASzot Mentioned that it’s not used in any of the Habitat 2.0 tasks. The RearrangeSim is only for Habitat 2.0 tasks, so it doesn’t impact that Habitat 1.0 tasks that use this.

So, we shouldn’t be using the _prev_sim_obs here.

Good catch.